### DISCUSSION TITLE
proper usage of context.eventHandler

### DISCUSSION THREAD

**Discussion by doug-precocity** [square/workflow-kotlin#335] [Q&A] [answered]
My initial impression was that `context.eventHandler{}` was just a shorthand for `context.actionSink.send( action{} )`
But in practice that seems not to be the case. I'm failing to see though why the former fails in the below example but the latter succeeds.  

```
override fun render(props:DiscoverProps, state:State, context:RenderContext):DiscoverScreen {
	return DiscoverRendering(
		onItemSelected = { info:SearchHit ->
			//this does not work
			//context.eventHandler { setOutput(DiscoverOutput(info)) }   
			
			//but this does
			context.actionSink.send( action { setOutput(DiscoverOutput(info)) } )
		}
	)
}
```
Is there some mistake I'm overlooking here, or are these meant to be used in different situations?

**Comment by rjrjr** [comment, marked as answer]
You don't need the outer lambda.

```kotlin
return DiscoverRendering(
 onItemSelected = context.eventHandler { info -> 
    setOutput(DiscoverOutput(info)) 
  }
)
```

**Reply by doug-precocity** [reply]
*smacks forehead*
